module/icp/algs/modes/gcm.c

```diff
@@ -67,7 +67,9 @@ gcm_mode_encrypt_contiguous_blocks(gcm_ctx_t *ctx, char *data, size_t length,
 		    (uint8_t *)ctx->gcm_remainder + ctx->gcm_remainder_len,
 		    length);
 		ctx->gcm_remainder_len += length;
-		ctx->gcm_copy_to = datap;
+		if (ctx->gcm_copy_to == NULL) {
```

Comment:
Does a similar change need to be made to the `/* Incomplete last block. */` check on lines 151-157?

Reply:
I don't think so. In the do loop at line 59 first we make sure that we have filled a complete block (line 85). On line 147 we have processed any remainder and advanced `datap` to the next block boundary. So `ctx->gcm_remainder` is processed and can be reused.  Now we check if there are less than `blocksize` bytes remaining. If not we reuse `ctx->gcm_remainder` and return. The next call will then hit line 65 or line 82 and so forth.

Reply:
Got it, that makes sense.